src/rgw/rgw_rest_s3.h

```diff
@@ -873,6 +873,83 @@ static inline int valid_s3_bucket_name(const string& name, bool relaxed=false)
   return 0;
 }
 
+class s3select;
+class RGWS3Select : public RGWGetObj_ObjStore_S3
+{
+
+private:
+  s3select *s3select_syntax;
+  std::string m_last_line;
+  bool m_previous_line;
+
+public:
+  unsigned int chunk_number;
+  u_int64_t m_processed_bytes;
+
+  typedef enum
```

Comment:
nit: no need for typedef enum, just enum or enum class

Reply:
also, prefer enum class